framework/src/loops/ComputeMortarFunctor.C

```diff
@@ -428,32 +470,25 @@ ComputeMortarFunctor::projectQPoints3d(const Elem * msm_elem,
     if (current_iterate < max_iterates)
     {
       // Transfer quadrature point from sub-element to element and store
-      q_pts[qp] = transform_qp(xi(0).value(), xi(1).value());
+      q_pts.push_back(transform_qp(xi(0).value(), xi(1).value()));
 
       // The following checks if quadrature point falls in correct domain.
       // On small mortar segment elements with very distorted elements this can fail, instead of
       // erroring simply truncate quadrature point, these points typically have very small
       // contributions to integrals
       if (primal_elem->type() == TRI3 || primal_elem->type() == TRI6)
       {
-        if (q_pts[qp](0) < 0 || q_pts[qp](1) < 0 || q_pts[qp](0) + q_pts[qp](1) > 1)
+        if (q_pts.back()(0) < 0 || q_pts.back()(1) < 0 || q_pts.back()(0) + q_pts.back()(1) > 1)
```

Comment:
I think it would be cleaner just to make a local `auto & qp = q_pts.back();` in the line above and then replace all the uses of `q_pts.back()` with it